### PR TITLE
Add OpenSearch sink doc

### LINKED ISSUE #2282: Document: feat(sink): refator es sink to support opensearch
This issue tracks the documentation update needed for the merged PR #16330.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/16330
Source PR Merged At: 2024-06-18T09:53:18Z

### PR DESCRIPTION
**Pull request by WanYixian**
## Info

- **Description**

  - Add OpenSearch sink doc

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/16330

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2282

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

**Comment by WanYixian** [comment]
Hi @xxhZs, please help check if all notes are applicable for OpenSearch, and add the prerequisite for OpenSearch if any, [like this in Elasticsearch](https://docs.risingwave.com/docs/current/sink-to-elasticsearch/#prerequisites), thanks!

File: docs/guides/sink-to-opensearch.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+|timestamp with time zone |text|
+|interval |text|
+|struct |object|
+|array |array|
+|JSONB|object (RisingWave's OpenSearch sink will send JSONB as a JSON string, and OpenSearch will convert it into an object)|
+
+:::note
+OpenSearch doesn't require users to explicitly `CREATE TABLE`. Instead, it infers the schema on-the-fly based on the first record ingested. For example, if a record contains a jsonb '{v1: 100}', v1 will be inferred as a long type. However, if the next record is '{v1: "abc"}', the ingestion will fail because "abc" is inferred as a string and the two types are incompatible.
+
+This behavior should be noted, or your data may be less than it should be. In terms of monitoring, you can check out Grafana, where there is a panel for all sink write errors.
```
Comment (emile-00) [inline docs/guides/sink-to-opensearch.md]:
```suggestion
This behavior may lead to missing records. For monitoring, see Grafana, where there is a panel for all sink write errors.
```

File: docs/guides/sink-to-opensearch.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+| `url`          | Required. URL of the OpenSearch REST API endpoint.|
+| `username`        | Optional. `opensearch` user name for accessing the OpenSearch endpoint. It must be used with `password`.|
+| `password`       | Optional. Password for accessing the OpenSearch endpoint. It must be used with `username`.|
+|`delimiter` | Optional. Delimiter for OpenSearch ID when the sink's primary key has multiple columns.|
+
+## Notes about primary keys and OpenSearch IDs
+
+The OpenSearch sink defaults to the `upsert` sink type. It does not support the `append-only` sink type.
+
+If you want to customize your OpenSearch ID, please specify it via the `primary_key` parameter. RisingWave will combine multiple primary key values into a single string with the delimiter you set, and use it as the OpenSearch ID.
```
Comment (emile-00) [inline docs/guides/sink-to-opensearch.md]:
```suggestion
To customize your OpenSearch ID, specify it via the `primary_key` parameter. RisingWave will combine multiple primary key values into a single string with the delimiter set, and use it as the OpenSearch ID.
```

**Review by xxhZs — APPROVED** [review]
lgtm

**Pull request merged**